### PR TITLE
MLflow Integration doc Update

### PR DESCRIPTION
**Pull request by michael-berk**
# Description
LlamaIndex is a supported flavor in MLlfow with the 2.15 release. This PR adds MLflow to the integration docs.

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [x] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [x] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by michael-berk** [comment]
@B-Step62 and @BenWilson2 for visbility.

File: docs/docs/module_guides/observability/index.md
Comment (B-Step62) [inline docs/docs/module_guides/observability/index.md:438]:
```suggestion
```

duplication

File: docs/docs/module_guides/observability/index.md
Comment (B-Step62) [inline docs/docs/module_guides/observability/index.md:464]: Apparently this page purely focus on observability. Can we replace this with normal llama-index querying without logging? I think there is a better place to put logging integration.

Comment (michael-berk) [reply docs/docs/module_guides/observability/index.md:452]: Yeah good point. Will definitely defer to maintainers (but will wait for comments before redoing the gif)

Comment (logan-markewich) [reply docs/docs/module_guides/observability/index.md:452]: I think as discussed on slack, there isn't really a better place to put this for now

**Comment by logan-markewich** [comment]
@michael-berk @B-Step62 I saw v2.15 of mlflow got released, is this safe to merge/wrap up?

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**
